src/Microsoft.AspNetCore.Rewrite/UrlRewrite/InputParser.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Rewrite.Internal;
+
+namespace Microsoft.AspNetCore.Rewrite.UrlRewrite
+{
+    /// <summary>
+    /// </summary>
+    public class InputParser
+    {
+        private const char Colon = ':';
```

Comment:
if you're parsing information from the xml, pass the line number so your exceptions can include that. Without that, if you have a big xml, finding the error is very hard